chromadb/api/configuration.py

```diff
@@ -268,19 +270,70 @@ def from_legacy_params(cls, params: Dict[str, Any]) -> Self:
         for name, value in params.items():
             if name not in old_to_new:
                 raise ValueError(f"Invalid legacy HNSW parameter name: {name}")
-            new_name = old_to_new[name]
-            parameters.append(ConfigurationParameter(name=new_name, value=value))
-        return cls(parameters=parameters)
+            parameters.append(
+                ConfigurationParameter(name=old_to_new[name], value=value)
+            )
+        return cls(parameters)
+
+
+class HNSWConfigurationInterface(HNSWConfigurationInternal):
+    def __init__(
+        self,
+        space: str = "l2",
+        ef_construction: int = 100,
+        ef_search: int = 10,
+        num_threads: int = cpu_count(),
+        M: int = 16,
+        resize_factor: float = 1.2,
+        batch_size: int = 1000,
+        sync_threshold: int = 100,
+    ):
+        parameters = [
+            ConfigurationParameter(name="space", value=space),
+            ConfigurationParameter(name="ef_construction", value=ef_construction),
+            ConfigurationParameter(name="ef_search", value=ef_search),
+            ConfigurationParameter(name="num_threads", value=num_threads),
+            ConfigurationParameter(name="M", value=M),
+            ConfigurationParameter(name="resize_factor", value=resize_factor),
+            ConfigurationParameter(name="batch_size", value=batch_size),
+            ConfigurationParameter(name="sync_threshold", value=sync_threshold),
+        ]
 
+        super().__init__(parameters=parameters)
 
-class CollectionConfiguration(Configuration):
-    """The configuration for a collection."""
+
+# Alias for user convenience
+HNSWConfiguration = HNSWConfigurationInterface
+
+
+class CollectionConfigurationInternal(ConfigurationInternal):
+    """Internal representation of the collection configuration. Used for validation and defaults."""
 
     definitions = {
         "hnsw_configuration": ConfigurationDefinition(
             name="hnsw_configuration",
-            validator=lambda value: isinstance(value, HNSWConfiguration),
+            validator=lambda value: isinstance(value, HNSWConfigurationInternal),
             is_static=False,
-            default_value=HNSWConfiguration(),
+            default_value=HNSWConfigurationInternal(),
         ),
     }
+
+
+class CollectionConfigurationInterface(CollectionConfigurationInternal):
```

Comment:
Can we add some basic tests from the api level that excercise the validation and static checking?

Reply:
Discussed offline: since the configuration is not consumed yet, and the user doesn't have direct access to it after creation, there aren't really API-level tests we can write here. However, unit tests are probably warranted, will add them to this PR.